### PR TITLE
Dennis nc news 6

### PR DESCRIPTION
**Pull request by DennisTockan**

### CODE REVIEW

File: endpoints.json [hunk 1]
```diff
@@ -44,5 +44,23 @@
...
+      "articles": [
+        {
+          "comment_id": 9,
+          "body": "Superficially charming",
+          "article_id": 1,
+          "author": "icellusedkars",
+          "votes": 0,
+          "created_at": "2020-01-01T03:08:00.000Z"
+        }
+      ]
```
Comment (Jodiech) [inline endpoints.json:62]: Nice to see you are updating your endpoints.json file 👍

File: __tests__/app.test.js [hunk 2]
```diff
@@ -43,4 +43,39 @@ describe("GET /api", ()=> {
...
   })
+})
+
+describe('GET /api/articles/:article_id/comments', () => {
+  test('200 accepts an article_id and responds with an array of comments for that given id ', () => {
+    return request(app)
+    .get("/api/articles/1/comments")
+    .expect(200)
+    .then(({body}) => {
+      const {comments} = body;
```
Comment (Jodiech) [inline __tests__/app.test.js:67]: Please add a length check before your forEach ❗ 
I'd also add in something to check the order of the comments - take a look at jest-sorted library for this 👍

File: __tests__/app.test.js [hunk 3]
```diff
@@ -43,4 +43,39 @@ describe("GET /api", ()=> {
...
+      comments.forEach((comment) => {
+        expect(comment).toHaveProperty("comment_id", expect.any(Number));
+        expect(comment).toHaveProperty("votes", expect.any(Number));
+        expect(comment).toHaveProperty("created_at", expect.any(String));
+        expect(comment).toHaveProperty("author", expect.any(String));
+        expect(comment).toHaveProperty("body", expect.any(String));
+        expect(comment).toHaveProperty("article_id", expect.any(Number));
+      })
+    })
+  })
```
Comment (Jodiech) [inline __tests__/app.test.js]: You're missing a test for a valid ID with no comments. What status would this be? You would also want this to respond with an empty array 👍

**Review by Jodiech — COMMENTED** [review]
Hey 👋 nice work! I would like you to add another test for when a valid ID is passed that has no comments. Please make changes and resubmit when you're ready 👍

File: __tests__/app.test.js [hunk 4]
```diff
@@ -49,6 +49,61 @@ describe("GET /api", () => {
...
+      expect(comments).toHaveLength(11); 
+      expect(comments).toBeSorted({key: 'created_at', descending: true});
+      comments.forEach((comment) => {
+        expect(comment).toHaveProperty("comment_id", expect.any(Number));
+        expect(comment).toHaveProperty("votes", expect.any(Number));
+        expect(comment).toHaveProperty("created_at", expect.any(String));
+        expect(comment).toHaveProperty("author", expect.any(String));
+        expect(comment).toHaveProperty("body", expect.any(String));
+        expect(comment).toHaveProperty("article_id", expect.any(Number));
+      })
```
Comment (Poonam-raj) [inline __tests__/app.test.js:77]: it's worth noting that you can definitely be more specific about what article_id value comes back, these comments will all be for the same article

File: models/article_id_comments.model.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    );
+    if (dbOutput.rows.length === 0){
+      return Promise.reject({ status: 404, message: "Not Found" });
+    }
+  };
+
+const doesArticleIdExist = await checkExists(article_id);
+  if (doesArticleIdExist) {
+    return doesArticleIdExist;
+  }
```
Comment (Poonam-raj) [inline models/article_id_comments.model.js:17]: Something to think about:
you don't necessarily need to check if the article exists in ALL cases. In which case do you know you need to do a check like this? (or what kind of result do you need to do this in)

**Review by Poonam-raj — APPROVED** [review]
Nice work Dennis, some thoughts left below but great work considering the feedback and making changes.

You're free to merge

**Pull request merged**